examples/algorithms/plot_dedensification.py

```diff
@@ -0,0 +1,102 @@
+"""
+=============
+Dedensification
+=============
+
+Example of dedensifying a graph
+"""
+
+import networkx as nx
+from networkx.algorithms import summarization
+from networkx.drawing.nx_agraph import graphviz_layout as layout
+import matplotlib.pyplot as plt
+
+plt.suptitle("Dedensification")
+
+original_graph = nx.DiGraph()
+white_nodes = ["1", "2", "3", "4", "5", "6"]
+red_nodes = ["A", "B", "C"]
+node_sizes = [250 for node in white_nodes + red_nodes]
+node_colors = []
+for node in white_nodes:
+    node_colors.append("white")
+for node in red_nodes:
+    node_colors.append("red")
+
+original_graph.add_nodes_from(white_nodes + red_nodes)
+original_graph.add_edges_from(
+    [
+        ("1", "C"),
+        ("1", "B"),
+        ("2", "C"),
+        ("2", "B"),
+        ("2", "A"),
+        ("3", "B"),
+        ("3", "A"),
+        ("3", "6"),
+        ("4", "C"),
+        ("4", "B"),
+        ("4", "A"),
+        ("5", "B"),
+        ("5", "A"),
+        ("6", "5"),
+        ("A", "6"),
+    ]
+)
+
+layers = (("1", "2", "3"), ("C", "B", "A", "6"), ("4", "5"))
+for index, layer in enumerate(layers):
+    for node in layer:
+        original_graph.nodes[node]["layer"] = index
+
+base_options = dict(with_labels=True, edgecolors="black")
+pos = nx.multipartite_layout(original_graph, subset_key="layer")
```

Comment:
The positions might look better if we're more prescriptive on the position. These are integer locations, but might be good enough.
```suggestion
# position nodes in a 3x3 grid 
lines = (("1", "2", "3"), ("A", "B", "C"), ("4", "5", "6"))
orig_pos = {n: (line, spot) for line, nodes in enumerate(lines) for spot, n in enumerate(nodes)}
```

Reply:
@dschult Thank you for the grid code.  I tried it and the figure produced tended to have lots of overlapping edges which made the figure harder to read.  Here are those results:

![dedensification_example_suggested](https://user-images.githubusercontent.com/3859635/95672686-02232600-0b71-11eb-8d0f-0d8ed0cc140e.png)

 I got the best results (least number of overlapping edges) by hard-coding the positions to match those in the paper, so I am going to include those `pos` values.  Here 
![dedensification_example_paper_replicated](https://user-images.githubusercontent.com/3859635/95672717-2b43b680-0b71-11eb-80f6-1d02b8ca8189.png)